Bonsai.Harp.CF/SyringePumpCommand.cs

```diff
@@ -0,0 +1,175 @@
+        {
+            get
+            {
+                switch(Type)
+                {
+                    case SyringePumpCommandType.EnableMotorDriver: return "Enables the motor on the Syringe Pump";
+                    case SyringePumpCommandType.DisableMotorDriver: return "Disables the motor on the Syringe Pump";
+                    case SyringePumpCommandType.StartProtocol: return "Start the configured protocol on the Syringe Pump";
+                    case SyringePumpCommandType.StopProtocol: return "Stop the running protocol on the Syringe Pump";
+                    case SyringePumpCommandType.SetDigitalOutputs: return "Set the state of digital output 0 using the input boolean value.";
+                    case SyringePumpCommandType.ClearDigitalOutputs: return "Set the state of digital output 1 using the input boolean value.";
+                    case SyringePumpCommandType.MotorMicrostep: return "Set the motor microstep value.";
+                    case SyringePumpCommandType.ProtocolNumberOfSteps: return "Set the number of steps to run in the protocol [1;65535]";
+                    case SyringePumpCommandType.ProtocolStepsPeriod: return "Set the period in ms between each step on the protocol [1;65535]";
+                    case SyringePumpCommandType.ProtocolFlowRate: return "Set the flow rate of the protocol [0.5;2000.0]";
+                    case SyringePumpCommandType.ProtocolVolume: return "Set the volume in uL of the protocol [0.5;2000.0]";
+                    case SyringePumpCommandType.ProtocolType: return "Set the type of the protocol. False for step-based and True to volume-based protocol.";
+                    case SyringePumpCommandType.CalibrationValue1: return "Set the calibration value 1 for protocol use.";
+                    case SyringePumpCommandType.CalibrationValue2: return "Set the calibration value 2 for protocol use.";
+
+                    default: return null;
+                }
+            }
+        }
+
+        protected override Expression BuildSelector(Expression expression)
+        {
+            switch (Type)
+            {
+                case SyringePumpCommandType.EnableMotorDriver:
+                    return Expression.Call(typeof(SyringePumpCommand), nameof(ProcessEnableMotorDriver), null);
+                case SyringePumpCommandType.DisableMotorDriver:
+                    return Expression.Call(typeof(SyringePumpCommand), nameof(ProcessDisableMotorDriver), null);
+                case SyringePumpCommandType.StartProtocol:
+                    return Expression.Call(typeof(SyringePumpCommand), nameof(ProcessStartProtocol), null);
+                case SyringePumpCommandType.StopProtocol:
+                    return Expression.Call(typeof(SyringePumpCommand), nameof(ProcessStopProtocol), null);
+                case SyringePumpCommandType.SetDigitalOutputs:
+                    if(expression.Type != typeof(byte)) { expression = Expression.Convert(expression, typeof(byte)); }
```

Comment:
Missing space between the `if` keyword and the parenthesis. There are more instances of this below, so probably easiest to just quickly format the file.